### PR TITLE
Promote bootstrap logger to public API

### PR DESCRIPTION
**Pull request by macsux**
Promote bootstrap logger to public API. Improve intellisense

### CODE REVIEW

File: src/Common/src/Common/Logging/BootstrapLoggerFactory.cs [hunk 1]
```diff
@@ -4,7 +4,12 @@
 
 namespace Steeltoe.Common.Logging;
 
-internal static class BootstrapLoggerFactory
+/// <summary>
+/// Provides access to logging infrastructure before service container is created. Any loggers created are upgraded with config settings after
```
Comment (ccheetham) [inline src/Common/src/Common/Logging/BootstrapLoggerFactory.cs:8]: Should "upgraded " be "updated"?

File: src/Common/src/Common/Logging/BootstrapLoggerFactory.cs [hunk 2]
```diff
@@ -4,7 +4,12 @@
...
+/// config subsystem and later DI subsystem are available.
+/// This class should only be used by components that need logging logging infrastructure before service container is available.
+/// </summary>
```
Comment (ccheetham) [inline src/Common/src/Common/Logging/BootstrapLoggerFactory.cs:11]: "service container" seems odd.  If it's a singleton, would Service Container be better?  Or perhaps "the service container."

File: src/Common/src/Common/Logging/UpgradableBootstrapLoggerFactory.cs [hunk 3]
```diff
@@ -12,7 +12,7 @@ namespace Steeltoe.Common.Logging;
 /// <summary>
 /// Allows early utilization of log infrastructure before log config is even read. Any providers spawned are instantly switched over to
 /// real log providers as the application utilization progresses.
-/// This class should only be used by components start are invoke before  logging infrastructure is build (prior to service container creation).
+/// This class should only be used by components that need logging logging infrastructure before service container is available.
```
Comment (ccheetham) [inline src/Common/src/Common/Logging/UpgradableBootstrapLoggerFactory.cs:15]: ditto above

File: src/Common/src/Common/Logging/BootstrapLoggerFactory.cs [hunk 4]
```diff
@@ -4,7 +4,12 @@
 
 namespace Steeltoe.Common.Logging;
 
-internal static class BootstrapLoggerFactory
+/// <summary>
+/// Provides access to logging infrastructure before service container is created. Any loggers created are updated with config settings after
+/// config subsystem and later DI subsystem are available.
+/// This class should only be used by components that need logging logging infrastructure before Service Container is available.
```
Comment (bart-vmware) [inline src/Common/src/Common/Logging/BootstrapLoggerFactory.cs:10]:
```suggestion
/// This class should only be used by components that need logging infrastructure before Service Container is available.
```

File: src/Common/src/Common/Logging/UpgradableBootstrapLoggerFactory.cs [hunk 5]
```diff
@@ -12,7 +12,7 @@ namespace Steeltoe.Common.Logging;
 /// <summary>
 /// Allows early utilization of log infrastructure before log config is even read. Any providers spawned are instantly switched over to
 /// real log providers as the application utilization progresses.
-/// This class should only be used by components start are invoke before  logging infrastructure is build (prior to service container creation).
+/// This class should only be used by components that need logging logging infrastructure before Service Container is available.
```
Comment (bart-vmware) [inline src/Common/src/Common/Logging/UpgradableBootstrapLoggerFactory.cs:15]:
```suggestion
/// This class should only be used by components that need logging infrastructure before Service Container is available.
```

**Review by bart-vmware — CHANGES_REQUESTED** [review]

**Review by bart-vmware — APPROVED** [review]

**Pull request merged**